tfx/examples/bert/cola/bert_cola_pipeline.py

```diff
@@ -0,0 +1,199 @@
+# limitations under the License.
+"""Cola dataset classification example using TFX."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+from typing import List, Text
+
+import absl
+import tensorflow_model_analysis as tfma
+
+from tfx.components.example_gen.import_example_gen.component import ImportExampleGen
+from tfx.components import Evaluator
+from tfx.components import ExampleValidator
+from tfx.components import Pusher
+from tfx.components import ResolverNode
+from tfx.components import SchemaGen
+from tfx.components import StatisticsGen
+from tfx.components import Trainer
+from tfx.components import Transform
+from tfx.components.trainer.executor import GenericExecutor
+from tfx.components.base import executor_spec
+from tfx.dsl.experimental import latest_blessed_model_resolver
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+from tfx.proto import example_gen_pb2
+from tfx.proto import pusher_pb2
+from tfx.proto import trainer_pb2
+from tfx.types import Channel
+from tfx.types.standard_artifacts import Model
+from tfx.types.standard_artifacts import ModelBlessing
+from tfx.utils.dsl_utils import tfrecord_input
+
+_pipeline_name = 'bert_cola'
+
+# This example assumes the utility function is in ~/bert/cola
+_bert_cola_root = os.path.join(os.environ['HOME'], 'bert', 'cola')
+_data_root = os.path.join(_bert_cola_root, 'data')
+# Python module file to inject customized logic into the TFX components. The
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file = os.path.join(_bert_cola_root, 'bert_cola_utils.py')
+# Path which can be listened to by the model server.  Pusher will output the
+# trained model here.
+_serving_model_dir = os.path.join(
+    _bert_cola_root,
+    'serving_model',
+    _pipeline_name)
+
+# Directory and data locations. This example assumes all of the
+# example code and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
+_pipeline_root = os.path.join(_tfx_root, 'pipelines', _pipeline_name)
+# Sqlite ML-metadata db path.
+_metadata_path = os.path.join(_tfx_root, 'metadata', _pipeline_name,
+                              'metadata.db')
+
+# Pipeline arguments for Beam powered Components.
+# TODO: changed direct_num_workers=0 when https://github.com
+# /tensorflow/text/issues/311 is resolved.
+_beam_pipeline_args = ['--direct_num_workers=1']
+
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text, serving_model_dir: Text,
+                     metadata_path: Text,
+                     beam_pipeline_args: List[Text]) -> pipeline.Pipeline:
+  """Implements the Bert classication on Cola dataset pipline with TFX."""
```

Comment:
sp: pipeline

(Alternatively: "Implements BERT classification on the Cola dataset using TFX")

Reply:
fixed